### PR TITLE
163 151 make difference between local storage and cloud storage

### PR DESCRIPTION
**Pull request by BartLucien**
Once Merged, this PR will allow the user to choose whether he wants to store the playlist online or only offline. He can also go in the editing of playlist to store a playlist online.
To do so, generation of id for the songs of a playlist and registering them have been implemented.

### CODE REVIEW

File: app/src/main/java/com/github/fribourgsdp/radio/FirestoreDatabase.kt [hunk 1]
```diff
@@ -50,6 +50,8 @@ class FirestoreDatabase(var refMake: FirestoreRef) : Database {
 
                     val pl = Playlist(playlist["playlistName"]!!,Genre.valueOf(playlist["genre"]!!))
                     pl.id = playlist["playlistId"]!!
+                    pl.savedOnline = true
+                    pl.savedLocally = false
```
Comment (nathanduchesne) [inline app/src/main/java/com/github/fribourgsdp/radio/FirestoreDatabase.kt:54]: Is the getUser function that contains this code only called when the user signs in with Google? What I am trying to understand is, is it possible that a user saves a playlist locally, then logs into Google and getUser is called, and his local playlist is overriden by the newly loaded playlists?

File: app/src/main/java/com/github/fribourgsdp/radio/FirestoreDatabase.kt [hunk 2]
```diff
@@ -192,12 +197,12 @@ class FirestoreDatabase(var refMake: FirestoreRef) : Database {
...
 
             val id = snapshot.getLong(keyID)!!
-            val nextId = (id + 1) % snapshot.getLong(keyMax)!!
+            val nextId = (id + number) % snapshot.getLong(keyMax)!!
 
             transaction.update(docRef, keyID, nextId)
 
             // Success
-            id
+            Pair(id,nextId)
```
Comment (nathanduchesne) [inline app/src/main/java/com/github/fribourgsdp/radio/FirestoreDatabase.kt:205]: I haven't found where the nextId element is used. Is it necessary to have this Pair? Is it to keep track of which id we're on?

**Review by nathanduchesne — COMMENTED** [review]
Once you answer all my questions it's all good for me!

File: app/src/androidTest/java/com/github/fribourgsdp/radio/PlaylistSongsFragmentTest.kt [hunk 3]
```diff
@@ -107,4 +116,83 @@ class PlaylistSongsFragmentTest {
...
+
+        val scenario = launchFragmentInContainer<MockPlaylistSongsFragment>(bundle)
+        scenario.use {
+
+            onView(withId(R.id.SaveOnlineButton)).perform(ViewActions.click())
+            onView(withId(R.id.SaveOnlineButton)).check(matches(isNotClickable()))
+        }
+
+        //launch the fragment and click on save
+    }*/
```
Comment (VictorSchneuwly) [inline app/src/androidTest/java/com/github/fribourgsdp/radio/PlaylistSongsFragmentTest.kt:152]: If you don't consider using it later, you should delete this.

File: app/src/main/java/com/github/fribourgsdp/radio/Database.kt [hunk 4]
```diff
@@ -46,6 +46,7 @@ interface Database {
      */
     fun getLobbyId() : Task<Long>
 
+    fun generateSongIds(number: Int) :Task<Pair<Long,Long>>
```
Comment (VictorSchneuwly) [inline app/src/main/java/com/github/fribourgsdp/radio/Database.kt:49]: Add the doc for next time ;)

**Review by VictorSchneuwly — COMMENTED** [review]
Here are my remarks for now, before the merge.

File: app/src/main/java/com/github/fribourgsdp/radio/Database.kt (hunk 4 above)
Comment (BartLucien) [reply app/src/main/java/com/github/fribourgsdp/radio/Database.kt:53]: done

**Review by sim0mo — APPROVED** [review]
Tested it and it works. It's good for me.

**Review by nathanduchesne — APPROVED** [review]

**Pull request merged**